src/hypothesis/strategies.py

```diff
@@ -651,13 +662,24 @@ def characters(whitelist_categories=None, blacklist_categories=None,
                 min_codepoint, max_codepoint
             )
         )
+    if all((whitelist_characters is not None,
+            min_codepoint is None,
+            max_codepoint is None,
+            whitelist_categories is None,
+            blacklist_categories is None,
+            )):
+        raise InvalidArgument(
+            'Cannot have just whitelist_characters="%s" alone, '
+            'it would have no effect' % (whitelist_characters,)
```

Comment:
It might be worth suggesting an alternative to the tune of "Perhaps you meant sampled_from(%r)"?

Also I think you want `%r` rather than `%s` here?